skyportal/handlers/api/galaxy.py

```diff
@@ -0,0 +1,104 @@
+from baselayer.app.access import permissions, auth_or_token
+
+import astropy.units as u
+import healpix_alchemy as ha
+
+from ..base import BaseHandler
+from ...models import DBSession, Galaxy
+
+
+class GalaxyCatalogHandler(BaseHandler):
+    @permissions(['System admin'])
+    async def post(self):
```

Comment:
Why async?

Reply:
I was thinking in case the database write was slow? In principle, this can run out of time right?

Reply:
Yes, unfortunately just making it async won't help; the inside needs to be made async too, and currently nothing inside it exposes async functionality.

Reply:
Ok I moved the galaxy write to be async.